### ISSUE TITLE
Submission for assignment 2

### ISSUE THREAD

**Issue by tfengaka** [tfengaka/df-frontend-2023#2]
Link deploy: https://assignment-2-df-frontend-2023.vercel.app

😂Task công ty dí quá nên em không có thời gian làm mấy cái dark mode =))

**Comment by ngolapnguyen** [comment]
Good work 👍

## Requirements

- [x] Render book list with pagination
- [x] Create book
- [x] Delete book
- [x] Search book by title
- [x] Standard use of React component & state
- [ ] Light/dark mode

Final result: ✅ passed 70% of requirements

## Feedbacks

1. Minor stuff, you might want to reduce the spacing:

<img width="563" alt="image" src="https://github.com/tfengaka/df-frontend-2023/assets/11691985/b1ca13dc-8837-49df-86fd-db1e982fffbe">

It's too wide & a bit hard to follow 😅

2. Naming is decent, but could be a bit better. Some nitpicks:

- `TableData` -> `BookTable` ("TableData" sounds like a set of data)
- `SearchItem` -> `SearchInput` (clearer on functionality of the component)
- `dataContext` -> `DataContext` (component name -> UpperCamelCase)
- ...

They are some of the things that confused me when I was reading your code. Even though it's trivial, it'll affect the overall readability of your code.

3. Good job moving the heavy logic into `useDataProvider` hook 👍 It's a good practice. I suggest you move it into the same folder as the DataContext, though, considering that the logic is specific to the context. Something like:

```
contexts/
  DataContext/
    DataContext.js
    hooks/
      useDataProvider.ts
```

It'll be more manage-able & easy to find when your project grows bigger.
